oauthenticator/azuread.py

```diff
@@ -96,6 +103,11 @@ async def authenticate(self, handler, data=None):
 
         return userdict
 
+    def load_user_groups(self, auth_state):
+        auth_user_state = auth_state["auth_state"]["user"]
+        return auth_user_state[self.user_groups_claim]
```

Comment:
Can we assume `auth_user_state` always include a claim named `groups`? Does it imply that you have requested the scope `groups` as well or not?

Reply:
Is the `load_user_groups` function an Authenticator base class function that is overridden?

Reply:
Hmmm, what is calling this function? I find no reference to it.

Reply:
Ah, this PR relies on https://github.com/jupyterhub/jupyterhub/pull/3548.

Reply:
@consideRatio - Unsure whether `auth_user_state` will return with a `groups`-claim.

The functionality in this PR requires `auth_users_state` to return some claim with a list of groups if `c.Authenticator.authenticator_managed_groups = True`. The claim name itself is configurable.

Reply:
The `groups`-claim is non standard for OIDC, but AAD makes it available as an optional claim
https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-optional-claims. 

In our case we extend our AAD's policy to make API calls to fetch and return group membership, as shown in this example: https://github.com/azure-ad-b2c/samples/tree/master/policies/rest-api-idp